### PR TITLE
ci: DRA enhancements

### PR DESCRIPTION
**Pull request by v1v**
### What

- Add who to contact with if a DRA failure or a package failure.
- Use releaseManager step
- Add DRA build failure analyser.

### Issue

Requires https://github.com/elastic/apm-pipeline-library/pull/1628

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -127,11 +143,19 @@ pipeline {
...
+                             type: 'snapshot',
+                             artifactsFolder: 'build/distributions',
+                             outputFile: env.DRA_OUTPUT)
+            }
+          }
+          post {
+            failure {
+              notifyStatus(analyse: true,
+                           file: "${BASE_DIR}/${env.DRA_OUTPUT}",
+                           subject: "[${env.REPO}@${env.BRANCH_NAME}] DRA failed.",
```
Comment (jlind23) [inline .ci/packaging.groovy:157]: Can we write entirely "Daily releasable artifact"?

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -127,11 +143,19 @@ pipeline {
...
+                           body: 'Contact the Release Platform team [#platform-release].')
```
Comment (jlind23) [inline .ci/packaging.groovy:158]: So if something fails, the only thing to do is to contact the release platform team right?

Comment (v1v) [reply .ci/packaging.groovy:158]: IIUC, the `DRA` component is a black box, so if the packages have been generated correctly, and the `DRA` stage fails then I'd say that's the way to go.

If the package generation fails earlier, then that's something to be investigated, though that's a unlikely scenario

**Review by ph — APPROVED** [review]

**Comment by v1v** [comment]
@mergifyio backport 8.2

**Comment by v1v** [comment]
https://github.com/Mergifyio backport 8.1

**Comment by v1v** [comment]
@Mergifyio backport 8.1

**Pull request merged**